Packs/ApiModules/Scripts/MicrosoftApiModule/MicrosoftApiModule_test.py

```diff
@@ -718,11 +723,50 @@ def test_generate_login_url():
     """
     from MicrosoftApiModule import generate_login_url
 
-    client = self_deployed_client()
+    client = self_deployed_client(grant_type=AUTHORIZATION_CODE)
 
     result = generate_login_url(client)
 
     expected_url = f'[login URL](https://login.microsoftonline.com/{TENANT}/oauth2/v2.0/authorize?' \
                    f'response_type=code&scope=offline_access%20https://graph.microsoft.com/.default' \
                    f'&client_id={CLIENT_ID}&redirect_uri=https://localhost/myapp)'
     assert expected_url in result.readable_output, "Login URL is incorrect"
+
+
+def test_get_access_token_auth_code_reconfigured(mocker, requests_mock):
+    # client = self_deployed_client(grant_type=AUTHORIZATION_CODE)
+    context = {'auth_code': AUTH_CODE, 'access_token': TOKEN,
+               'valid_until': 3605, 'current_refresh_token': REFRESH_TOKEN}
+
+    mocker.patch.object(demisto, 'getIntegrationContext', return_value=context)
+    mocker.patch.object(demisto, 'setIntegrationContext')
+
+    tenant_id = TENANT
+    client_id = CLIENT_ID
+    client_secret = CLIENT_SECRET
+    base_url = BASE_URL
+    auth_code = 'reconfigured_auth_code'
+    redirect_uri = REDIRECT_URI
+    resource = None
+    ok_codes = OK_CODES
+    grant_type = AUTHORIZATION_CODE
+
+    client = MicrosoftClient(self_deployed=True, tenant_id=tenant_id, auth_id=client_id, enc_key=client_secret,
+                             grant_type=grant_type, auth_code=auth_code, redirect_uri=redirect_uri,
+                             resource=resource, base_url=base_url, verify=True, proxy=False, ok_codes=ok_codes)
+
+    requests_mock.post(
+        APP_URL,
+        json={'access_token': TOKEN, 'expires_in': '3600'})
+
+    body = {
+        'client_id': CLIENT_ID,
+        'client_secret': CLIENT_SECRET,
+        'redirect_uri': REDIRECT_URI,
+        'grant_type': AUTHORIZATION_CODE,
+        'code': 'reconfigured_auth_code',
```

Comment:
```suggestion
        'code': auth_code,
```

Reply:
Actually, this is the point of the PR we want to change the auth code (making it different from the integration context so that integration is forced to create a new access token